### PR TITLE
Update the component list in Kyma cluster and local installation

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Updated the list of components installed as a part of Kyma (cluster) and Kyma Lite (local) packages.
- Added a CAUTION to Serverless v2 docs specifying Namespace and certificate requirements.

**Related issue(s)**
See also #7265

### CODE REVIEW

**Comment by michal-hudy** [comment]
>**CAUTION:** LGTM

File: docs/kyma/04-01-overview.md [hunk 1]
```diff
@@ -5,30 +5,48 @@ type: Installation
...
+| `tekton-pipelines` | ⛔️ | ⛔️ |
+| `knative-serving-init` | ✅ | ✅ |
+| `knative-serving` | ✅ | ✅ |
+| `knative-eventing` | ✅ | ✅ |
+| `knative-eventing-channel-kafka` | ⛔️ | ⛔️ |
+| `knative-eventing-channel-kafka-tests` | ⛔️ | ⛔️ |
+| `dex` | ✅ | ✅ |
+| `ory` | ✅ | ✅ |
+| `api-gateway` | ✅ | ✅ |
+| `function-controller-init` | ⛔️ | ⛔️ |
```
Comment (magicmatatjahu) [inline docs/kyma/04-01-overview.md]: `function-controller-init` and `function-controller` are installed after `knative-eventing-channel-kafka-tests`. Please update table :)

Comment (magicmatatjahu) [reply docs/kyma/04-01-overview.md]: Forget about this comment.

Comment (kazydek) [reply docs/kyma/04-01-overview.md]: As discussed, this order is already based on changes in this PR: https://github.com/kyma-project/kyma/pull/7244

**Pull request merged**